src/middlewares/validation.ts

```diff
@@ -480,6 +480,21 @@ const isProductExistById = async (req: Request, res: Response, next: NextFunctio
     return res.status(httpStatus.INTERNAL_SERVER_ERROR).json({ status: httpStatus.INTERNAL_SERVER_ERROR, message: error.message });
   }
 }
+const isProductExistToWishlist = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+     const product = await productRepositories.findProductToWishListById(req.params.id);

```

Comment:
I think this is bad way of checking if user already have this product to his wishList, you should check both productId and userId.
because by this you will get true if there is any buyer ever added this to his wishList.